### ISSUE TITLE
keras lstm layer produces different results in tf 1 and tf 2 due to different default activation

### ISSUE THREAD

**Issue by VincentYu68** [tensorflow/tensorflow#47495]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- TensorFlow version:
1. TF 1.0: v1.15.4-39-g3db52be7be8 1.15.5
2. TF 2.0: v2.4.0-49-g85c8b2a817f 2.4.1

**Describe the current behavior**
The LSTM layer in tf 1.0 uses "hard_sigmoid" as the default recurrent activation: https://github.com/tensorflow/tensorflow/blob/v1.15.0/tensorflow/python/keras/layers/recurrent.py#L2489

while it uses  "sigmoid" in tf 2.0: https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/keras/layers/recurrent_v2.py#L1077

As a result, when I try to execute the same code (without specifying recurrent_activation flag explicitly) in tf 1 and 2, I get slightly different results and took quite a while to find the cause.

**Describe the expected behavior**
Same code should produce the same result in both versions. Or some prompt should be given for easier debugging.

**Comment by Saduf2019** [comment]
@VincentYu68 

1. There is no suppport for tf 1.x now, there is support only for 2.x.
2. I ran the code shared, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/2a200017506a101be4189ac992cef9bf/untitled556.ipynb)
